superset-frontend/src/views/CRUD/data/dataset/AddDataset/index.tsx

```diff
@@ -91,22 +90,22 @@ export default function AddDataset() {
       })
     : undefined;
 
-  const getDatasetsList = async () => {
-    await UseGetDatasetsList(queryParams)
-      .then(json => {
-        setDatasets(json?.result);
-      })
-      .catch(error =>
-        logging.error('There was an error fetching dataset', error),
-      );
-  };
+  const { getDatasetsList, datasets, datasetNames } =
+    useGetDatasetsList(queryParams);
 
   useEffect(() => {
     if (dataset?.schema) {
       getDatasetsList();
     }
   }, [dataset?.schema]);
 
+  const id = window.location.pathname.split('/')[2];
```

Comment:
Because we've defined the path for this page as `/dataset/:datasetId` in `routes.tsx`, React Router will parse out that param for us if we use [`useParams`](https://v5.reactrouter.com/web/api/Hooks/useparams) hook:
```suggestion
  const { datasetId: id } = useParams<{ datasetId: string }>();
```

Reply:
Fixed in [`this commit`](https://github.com/apache/superset/pull/22043/commits/41fbaeb11bfd574a84a24732cadd2dae25e6f39d).